utils/base_class.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+import requests
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.select import Select
+
+@pytest.mark.usefixtures("setup")
+class BaseClass:
+    _url:str
+
+    def __init__(self, driver,url=None):
+        self.driver = driver
+        self.wait = WebDriverWait(self.driver, 10)
+        self.url = url or self._url
```

Comment:
please rename and inherit abstract class (abc.ABC) and make sure base classes are never instantiated

```suggestion

import abc

class BasePage(abc.ABC):
    _url:str

    def __init__(self, driver,url=None):
        self.driver = driver
        self.wait = WebDriverWait(self.driver, 10)
        self.url = url or self._url
```

Reply:
fixed